DESIGN.rst

```diff
@@ -0,0 +1,53 @@
+============
```

Comment:
Starting reading the documentation, it wasn't clear what was its goal. maybe we could add a line explaining what this document is about.

Reply:
oh I guess it will be in the Introduction 🤦